### PR TITLE
Problem: need hardfork to adjust feemarket parameters

### PR DESCRIPTION
**Pull request by yihuang**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## UNRELEASED
+
+- [#]() Support hard-fork style upgrades.
```
Comment (yihuang) [inline CHANGELOG.md:5]:
```suggestion
- [#1259](https://github.com/crypto-org-chain/cronos/pull/1259) Use a hard-fork style upgrade to adjust feemarket parameters.
```

**Review by mmsqe — APPROVED** [review]

File: app/forks.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+// Fork defines a struct containing the requisite fields for a non-software upgrade proposal
+// Hard Fork at a given height to implement.
+// There is one time code that can be added for the start of the Fork, in `BeginForkLogic`.
+// Any other change in the code should be height-gated, if the goal is to have old and new binaries
+// to be compatible prior to the upgrade height.
+//
+// Adapted from osmosis: https://github.com/osmosis-labs/osmosis/blob/057192c2c0949fde5673a5f314bf41816f808fd9/app/upgrades/types.go#L40
+type Fork struct {
+	// Upgrade version name, for the upgrade handler, e.g. `v7`
+	UpgradeName string
```
Comment (thomas-nguy) [inline app/forks.go:14]: is it used anywhere?

**Review by thomas-nguy — APPROVED** [review]

File: app/forks.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		params.MinGasPrice = sdk.NewDecWithPrec(10000, 9)
+		app.FeeMarketKeeper.SetParams(ctx, params)
+	},
+}
+
+// BeginBlockForks is intended to be ran in a chain upgrade.
+func BeginBlockForks(ctx sdk.Context, app *App) {
+	for _, fork := range Forks {
+		if ctx.BlockHeight() == fork.UpgradeHeight && ctx.ChainID() == fork.UpgradeChainId {
+			fork.BeginForkLogic(ctx, app)
```
Comment (JayT106) [inline app/forks.go:42]: will it panic in the function when the fork logic failed? otherwise LGTM.

Comment (yihuang) [reply app/forks.go:33]: the parameter set operation will panic internally and don't return an error.

File: app/forks.go (hunk 2 above)
Comment (yihuang) [reply app/forks.go:19]: no, just for human reader

**Pull request merged**